File: Binary Search.java
class Solution
{
public int search(int[] nums, int target)
{
int n = nums.length;
int l = 0, r = n - 1, mid;
while(l <= r){
mid = (l + r) / 2;
if(nums[mid] > target)
r = mid - 1;
if(nums[mid] < target)
l = mid + 1;
if(nums[mid] == target)
return mid;
}
return -1;
}
}

File: Binary Search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
     
      int s = 0;
      int e = nums.size() - 1;
       int mid = s+(e-s)/2;
       while(s<= e){

       if (nums[mid] == target){
           return mid;
       }
        if (nums[mid]> target){
            e= mid -1;

        }
        else{
            s = mid +1 ;

        }

        mid = s+ (e-s)/2;







        
    }return -1;
    
       }
};
